File: WebAPI/FMS.Service/DTOs/AccountStatement.cs
using FMS.Entity.Context.Models;
using System;
using System.Collections.Generic;
using System.Text;

namespace FMS.Service.DTOs
{
    public class AccountStatement
    {
        // Account
        public int AccountId { get; set; }
        public int AccountNumber { get; set; }
        public AccountType AccountType { get; set; }
        public decimal LastBalance { get; set; }

        // Bank
        public int BankId { get; set; }
        public string BankName { get; set; }

        // Transactions
        public List<Transaction> Transactions { get; set; }
    }

    public class Transaction
    {
        public int BankTransactionId { get; set; }
        public int PayeeId { get; set; }
        public string PayeeName { get; set; }
        public PayeeType PayeeType { get; set; }
        public decimal AmountPaid { get; set; }
        public DateTime TransactionDate { get; set; }
        public TransactionStatus TransactionStatus { get; set; }
        public decimal CurrentBalance { get; set; }
        public decimal LastBalance { get; set; }
        
        // decides either In or Out
        public TransactionType TransactionType { get; set; }
        public string RefCode { get; set; }
        public int SourceId { get; set; }
        public string SourceName { get; set; }
    }
}

File: Angular/src/app/models/accountStatement.ts
import Transaction from "./transaction";

export default class AccountStatement {   
    accountId: number;
    accountNumber: number;
    accountType: number;
    accountTypeDisplay: string;
    lastBalance: number;
    bankId: number;
    bankName: string;
    transactions: Transaction[];

    groupBy: string;
    // '',transactionDate, payeeName

    // summary report
    totalIn: number;
    totalOut: number;
    payeeTypeSummaryDisplay: string;
    startDate: Date;
    endDate: Date;
}